exercise-src/react-native/16-security/01-solution/src/services/auth/AuthProvider.tsx

```diff
@@ -17,35 +17,72 @@ export interface AuthProviderProps {
 }
 
 const AuthProvider: React.FC<AuthProviderProps> = ({ children }) => {
+  const [error, setError] = useState<Error | undefined>()
```

Comment:
For `useState`, if you give it a non-undefined type then don't supply a default value, the final type will get `| undefined` automatically. (IE: `| undefined` on this line isn't necessary. It certainly doesn't hurt tho.)